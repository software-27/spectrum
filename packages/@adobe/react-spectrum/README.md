# @adobe/react-spectrum

This package allows you to easily import any component published under @react-spectrum. Think of it as a helpful alias,
with each component "pinned" to a specific version.

Components are updated sporadically, so if you require a change from a version of a component that is not yet included
in this package, you are able to install the specific version and then update your import statement to use the
component from the @react-spectrum scope. It's worth noting that this may result in increased file size.

## Naming conventions

Any component published under the @react-spectrum scope will be available at the same path from @adobe/react-spectrum.
For example, if you want to use `Checkbox` from @react-spectrum this is what you'd be importing:

```js
import {Checkbox} from '@react-spectrum/checkbox';
```

To import the same component from @adobe/react-spectrum, it would look like this:

```js
import {Checkbox} from '@adode/react-spectrum/checkbox';
```

This convention will be consistent for all packages to make it easy to switch between @react-spectrum and
@adobe/react-spectrum.

## Component versions

| Component                          | Version        |
| ---------------------------------- | -------------- |
| @react-spectrum/actiongroup        | 3.0.0-alpha.1  |
| @react-spectrum/breadcrumbs        | 3.0.0-alpha.1  |
| @react-spectrum/button             | 3.0.0-rc.2     |
| @react-spectrum/buttongroup        | 3.0.0-alpha.1  |
| @react-spectrum/checkbox           | 3.0.0-rc.2     |
| @react-spectrum/dialog             | 3.0.0-alpha.1  |
| @react-spectrum/divider            | 3.0.0-rc.2     |
| @react-spectrum/form               | 3.0.0-rc.2     |
| @react-spectrum/icon               | 3.0.0-rc.2     |
| @react-spectrum/illustratedmessage | 3.0.0-alpha.1  |
| @react-spectrum/image              | 3.0.0-alpha.1  |
| @react-spectrum/label              | 3.0.0-rc.2     |
| @react-spectrum/layout             | 3.0.0-alpha.1  |
| @react-spectrum/link               | 3.0.0-alpha.1  |
| @react-spectrum/listbox            | 3.0.0-alpha.1  |
| @react-spectrum/menu               | 3.0.0-alpha.1  |
| @react-spectrum/meter              | 3.0.0-rc.2     |
| @react-spectrum/overlays           | 3.0.0-alpha.1  |
| @react-spectrum/picker             | 3.0.0-alpha.1  |
| @react-spectrum/progress           | 3.0.0-rc.2     |
| @react-spectrum/provider           | 3.0.0-rc.2     |
| @react-spectrum/radio              | 3.0.0-rc.2     |
| @react-spectrum/searchfield        | 3.0.0-rc.2     |
| @react-spectrum/statuslight        | 3.0.0-rc.2     |
| @react-spectrum/switch             | 3.0.0-rc.2     |
| @react-spectrum/textfield          | 3.0.0-rc.2     |
| @react-spectrum/theme-default      | 3.0.0-rc.2     |
| @react-spectrum/utils              | 3.0.0-rc.2     |
| @react-spectrum/view               | 3.0.0-alpha.1  |
| @react-spectrum/well               | 3.0.0-rc.2     |
