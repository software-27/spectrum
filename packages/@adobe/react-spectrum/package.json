{
  "name": "@adobe/react-spectrum",
  "version": "3.0.0-rc.0",
  "description": "Spectrum UI components in React",
  "license": "Apache-2.0",
  "private": true,
  "main": "dist/main.js",
  "module": "dist/module.js",
  "types": "dist/types.d.ts",
  "source": "src/index.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "scripts": {
    "prepublishOnly": "mkdir -p dist; cp src/index.ts dist/types.d.ts; cp src/index.ts dist/module.js; babel --root-mode upward src/index.ts -o dist/main.js"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/adobe-private/react-spectrum-v3"
  },
  "dependencies": {
    "@react-spectrum/actiongroup": "^3.0.0",
    "@react-spectrum/breadcrumbs": "^3.0.0",
    "@react-spectrum/button": "^3.0.0",
    "@react-spectrum/buttongroup": "^3.0.0",
    "@react-spectrum/checkbox": "^3.0.0",
    "@react-spectrum/dialog": "^3.0.0",
    "@react-spectrum/divider": "^3.0.0",
    "@react-spectrum/form": "^3.0.0",
    "@react-spectrum/icon": "^3.0.0",
    "@react-spectrum/illustratedmessage": "^3.0.0",
    "@react-spectrum/image": "^3.0.0",
    "@react-spectrum/label": "^3.0.0",
    "@react-spectrum/layout": "^3.0.0",
    "@react-spectrum/link": "^3.0.0",
    "@react-spectrum/listbox": "^3.0.0",
    "@react-spectrum/menu": "^3.0.0",
    "@react-spectrum/meter": "^3.0.0",
    "@react-spectrum/overlays": "^3.0.0",
    "@react-spectrum/picker": "^3.0.0",
    "@react-spectrum/progress": "^3.0.0",
    "@react-spectrum/provider": "^3.0.0",
    "@react-spectrum/radio": "^3.0.0",
    "@react-spectrum/searchfield": "^3.0.0",
    "@react-spectrum/statuslight": "^3.0.0",
    "@react-spectrum/switch": "^3.0.0",
    "@react-spectrum/text": "^3.0.0",
    "@react-spectrum/textfield": "^3.0.0",
    "@react-spectrum/theme-dark": "^3.0.0",
    "@react-spectrum/theme-default": "^3.0.0",
    "@react-spectrum/view": "^3.0.0",
    "@react-spectrum/well": "^3.0.0"
  }
}
