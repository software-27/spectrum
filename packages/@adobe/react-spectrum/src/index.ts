/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

export * from '@react-spectrum/actiongroup';
export * from '@react-spectrum/breadcrumbs';
export * from '@react-spectrum/button';
export * from '@react-spectrum/buttongroup';
export * from '@react-spectrum/checkbox';
export * from '@react-spectrum/dialog';
export * from '@react-spectrum/divider';
export * from '@react-spectrum/form';
export * from '@react-spectrum/icon';
export * from '@react-spectrum/illustratedmessage';
export * from '@react-spectrum/image';
export * from '@react-spectrum/label';
export * from '@react-spectrum/layout';
export * from '@react-spectrum/link';
export * from '@react-spectrum/listbox';
export * from '@react-spectrum/menu';
export * from '@react-spectrum/meter';
export * from '@react-spectrum/overlays';
export * from '@react-spectrum/picker';
export * from '@react-spectrum/progress';
export * from '@react-spectrum/provider';
export * from '@react-spectrum/radio';
export * from '@react-spectrum/searchfield';
export * from '@react-spectrum/statuslight';
export * from '@react-spectrum/switch';
export * from '@react-spectrum/text';
export * from '@react-spectrum/textfield';
export {theme as darkTheme} from '@react-spectrum/theme-dark';
export {theme as defaultTheme} from '@react-spectrum/theme-default';
export * from '@react-spectrum/view';
export * from '@react-spectrum/well';
