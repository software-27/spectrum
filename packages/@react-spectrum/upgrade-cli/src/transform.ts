import path from 'node:path';
import {run as jscodeshift} from 'jscodeshift/src/Runner.js';

const transformPath = path.join(__dirname, 'codemods', 'codemod.js');

interface Options {
  componentSubset?: string,
  path?: string,
  ignorePattern?: string,
  parser?: string
}

export async function transform(options: Options) {
  let {
    path: filePath = '.',
    ...rest
  } = options;

  return await jscodeshift(transformPath, [filePath], rest);
}
