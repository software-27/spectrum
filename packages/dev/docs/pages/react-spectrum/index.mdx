<!-- Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. -->

import {Layout} from '@react-spectrum/docs';
export default Layout;

import {Highlights, Hero} from '@react-spectrum/docs';
import heroNarrow from 'url:../assets/hero-narrow.png';
import heroNarrow2x from 'url:../assets/hero-narrow@2x.png';
import heroWide from 'url:../assets/hero-wide.png';
import heroWide2x from 'url:../assets/hero-wide@2x.png';

# React Spectrum

A React implementation of Spectrum, Adobeâ€™s design system.

<Hero
  narrow={heroNarrow}
  narrow2x={heroNarrow2x}
  wide={heroWide}
  wide2x={heroWide2x}
  alt="React Spectrum hero image" />

<Highlights items={[
  {
    title: 'Accessible',
    description: 'React Spectrum components are designed with accessibility in mind, including full screen reader and keyboard navigation support.',
  },
  {
    title: 'Adaptive',
    description: 'React Spectrum components are designed to work with mouse, touch, and keyboard interactions. Built with responsive design principles to deliver a great experience no matter the device.',
  },
  {
    title: 'Themeable',
    description: 'React Spectrum supports custom themes to match your brand, including automatic support for dark mode.'
  },
  {
    title: 'Internationalized',
    description: 'React Spectrum components are designed to support over 30 languages, including support for right to left, date and number formatting, and more.'
  }
]} />
