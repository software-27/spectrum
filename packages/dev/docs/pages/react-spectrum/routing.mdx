{/* Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {Layout} from '@react-spectrum/docs';
export default Layout;

import {Keyboard} from '@react-spectrum/text';

---
category: Concepts
keywords: [router, links, frameworks, next.js, remix, react-router]
order: 5
---

# Client Side Routing

Many React Spectrum components support rendering as HTML links. This page discusses how to set up your app to integrate React Spectrum links with your framework or client side router.

## Introduction

React Spectrum components such as [Menu](Menu.html), [Tabs](Tabs.html), [TableView](TableView.html), and many others support rendering elements as links that perform navigation when the user interacts with them. Each component that supports link behavior accepts the `href` prop, which causes the component to render an `<a>` element. Other link DOM props such as `target` and `download` are also supported.

By default, links perform native browser navigation when they are interacted with. However, many apps and frameworks use client side routers to avoid a full page reload when navigating between pages. The [Provider](Provider.html) component can configure all React Spectrum components within it to navigate using the client side router you provide. Set this up once in the root of your app, and any React Spectrum component with the `href` prop will automatically navigate using your router.

Note that external links to different origins will not trigger client side routing, and will use native browser navigation. Additionally, if the link has a [target](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target) other than `"_self"`, uses the [download](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#download) attribute, or the user presses modifier keys such as <Keyboard>Command</Keyboard> or <Keyboard>Alt</Keyboard> to change the default behavior, browser native navigation will occur instead of client side routing.

## Provider setup

The `Provider` component accepts a `router` prop, which can be set to a router object. This should include a `navigate` function received from your router for performing a client side navigation programmatically. The following example shows the general pattern. Framework-specific examples are shown below.

```tsx
import {Provider, defaultTheme} from '@adobe/react-spectrum';

function App() {
  let navigate = useNavigateFromYourRouter();

  return (
    <Provider theme={defaultTheme} router={{navigate}}>
      {/* ... */}
    </Provider>
  );
}
```

### React Router

The [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook from `react-router-dom` returns a `navigate` function you can pass to `Provider`. Ensure that the component that calls `useNavigate` and renders `Provider` is inside the router component (e.g. `BrowserRouter`) so that it has access to React Router's internal context. The React Router `<Routes>` element should also be defined inside React Spectrum's `<Provider>` so that links inside the rendered routes have access to the router.

```tsx
import {BrowserRouter, useNavigate} from 'react-router-dom';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

function App() {
  let navigate = useNavigate();

  return (
    <Provider theme={defaultTheme} router={{navigate}}>
      {/* Your app here... */}
      <Routes>
        <Route path="/" element={<HomePage />} />
        {/* ... */}
      </Routes>
    </Provider>
  );
}

<BrowserRouter>
  <App />
</BrowserRouter>
```

### Next.js

#### App router

The [useRouter](https://nextjs.org/docs/app/api-reference/functions/use-router) hook from `next/navigation` returns a router object that can be used to perform navigation. `Provider` should be rendered from a client component at the root of each page or layout that includes React Spectrum components. You can create a new client component for this, or combine it with other top-level providers as described in the [Next.js docs](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#using-context-providers).

```tsx
// app/provider.tsx
"use client";

import {useRouter} from 'next/navigation';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

export function ClientProviders({children}) {
  let router = useRouter();

  return (
    <Provider theme={defaultTheme} router={{navigate: router.push}}>
      {children}
    </Provider>
  );
}
```

Then, in your page or layout server component, wrap your app in the `ClientProviders` component that you defined.

```tsx
// app/layout.tsx
import {ClientProviders} from './provider';

export default function RootLayout({children}) {
  return (
    <html>
      <body>
        <ClientProviders>{children}</ClientProviders>
      </body>
    </html>
  );
}
```

#### Pages router

The [useRouter](https://nextjs.org/docs/pages/api-reference/functions/use-router) hook from `next/router` returns a router object that can be used to perform navigation. `Provider` should be rendered at the root of each page that includes React Spectrum components, or in `pages/_app.tsx` to add it to all pages.

```tsx
// pages/_app.tsx
import type {AppProps} from 'next/app';
import {useRouter} from 'next/router';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

export default function MyApp({Component, pageProps}: AppProps) {
  let router = useRouter();

  return (
    <Provider theme={defaultTheme} router={{navigate: router.push}}>
      <Component {...pageProps} />
    </Provider>
  );
}
```

### Remix

Remix uses React Router under the hood, so the same [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook described above also works in Remix apps. `Provider` should be rendered at the root of each page that includes React Spectrum components, or in `app/root.tsx` to add it to all pages. See the [Remix docs](https://remix.run/docs/en/main/file-conventions/root) for more details.

```tsx
// app/root.tsx
import {useNavigate, Outlet} from '@remix-run/react';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

export default function App() {
  let navigate = useNavigate();

  return (
    <html lang="en">
      <head>
        {/* ... */}
      </head>
      <body>
        <Provider theme={defaultTheme} router={{navigate}}>
          <Outlet />
        </Provider>
        {/* ... */}
      </body>
    </html>
  );
}
```
