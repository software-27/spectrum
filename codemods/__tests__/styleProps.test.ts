// @ts-ignore
import {defineSnapshotTest} from 'jscodeshift/dist/testUtils';
import transform from '../codemod';

const test = (name: string, input: string) => {
  defineSnapshotTest(transform, {}, input, name);
};

test('Handles dimensions', `
import {Button} from '@adobe/react-spectrum';

<Button marginX="size-200" marginY={20} width="100%">Test</Button>
`);

test('Handles renamed components', `
import {Button as RSPButton} from '@adobe/react-spectrum';

<RSPButton marginX="size-200" marginY={20} width="100%">Test</RSPButton>
`);

test('Adds a comment when a variable is used', `
import {Button} from '@adobe/react-spectrum';

<Button marginX={margin}>Test</Button>
`);

test('Adds a comment when a spread is used', `
import {Button} from '@adobe/react-spectrum';

<Button marginX="size-200" {...otherProps}>Test</Button>
`);

test('Handles Flex', `
import {Flex} from '@adobe/react-spectrum';

<Flex direction="column" alignItems="center" gap="size-200" wrap>Test</Flex>
`);

test('Handles Grid', `
import {Grid} from '@adobe/react-spectrum';

<Grid
  areas={[
    'header  header',
    'sidebar content',
    'footer  footer'
  ]}
  columns={['1fr', '3fr']}
  rows={['size-1000', 'auto', 'size-1000']}
  height="size-6000"
  gap="size-100">
  Test
</Grid>
`);

test('Handles View', `
import {View} from '@adobe/react-spectrum';

<View
  borderWidth="thin"
  borderColor="dark"
  borderRadius="medium"
  padding="size-250"
  backgroundColor="blue-600"
  flex>
  Test
</View>
`);

test('Handles View colorVersion={5}', `
import {View} from '@adobe/react-spectrum';

<View
  borderWidth="thin"
  borderColor="gray-300"
  borderRadius="medium"
  padding="size-250"
  backgroundColor="blue-600"
  colorVersion={5}>
  Test
</View>
`);

test('Handles View colorVersion={6}', `
import {View} from '@adobe/react-spectrum';

<View
  borderWidth="thin"
  borderColor="gray-300"
  borderRadius="medium"
  padding="size-250"
  backgroundColor="blue-600"
  colorVersion={6}>
  Test
</View>
`);

test('Handles responsive style props', `
import {TextField} from '@adobe/react-spectrum';

<TextField label="Name" width={{ base: 'size-2000', L: 'size-5000' }} />
`);

test('Handles UNSAFE_style', `
import {View} from '@adobe/react-spectrum';

<View
  UNSAFE_style={{
    color: isErrorState ? 'var(--spectrum-red-900)' : 'var(--spectrum-gray-900)',
    borderWidth: border === 'thin' ? 'var(--spectrum-alias-border-size-thin)' : 'var(--spectrum-alias-border-size-thick)',
    borderRadius: isRounded(props) ? 4 : 0,
    display: 'flex',
    gap: 'var(--spectrum-global-dimension-size-200)',
    fontSize: 'var(--spectrum-global-dimension-font-size-100)',
    fontWeight: 'var(--spectrum-global-font-weight-bold)',
    gridTemplateAreas: '"foo bar" "baz qux"',
    transitionDuration: 'var(--spectrum-global-animation-duration-200)'
  }}>
  Test
</View>
`);

test('Bails on UNSAFE_style if it contains a spread', `
import {View} from '@adobe/react-spectrum';

<View
  UNSAFE_style={{
    color: 'var(--spectrum-gray-900)',
    ...otherStyles
  }}>
  Test
</View>
`);

test('Adds a comment to UNSAFE_className', `
import {Button} from '@adobe/react-spectrum';

<Button UNSAFE_className="hi">
  Test
</Button>
`);

test('Converts UNSAFE_className to className on View', `
import {View} from '@adobe/react-spectrum';

<View UNSAFE_className="hi">
  Test
</View>
`);

test('Merges UNSAFE_className with macro className', `
import {View} from '@adobe/react-spectrum';

<View UNSAFE_className="hi" marginX="size-200">
  Test
</View>
`);

test('Handles responsive style props', `
import {TextField} from '@adobe/react-spectrum';

<TextField label="Name" width={{ base: 'size-2000', L: 'size-5000' }} />
`);
