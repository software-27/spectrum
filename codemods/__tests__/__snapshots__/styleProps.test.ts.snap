// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds a comment to UNSAFE_className 1`] = `
"import { Button } from "@react-spectrum/rainbow";

<Button // TODO: check this UNSAFE_className
UNSAFE_className="hi">
  Test
</Button>"
`;

exports[`Adds a comment when a spread is used 1`] = `
"import { Button } from "@react-spectrum/rainbow";
import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<Button
  // TODO: check this spread for style props
  {...otherProps}
  css={style({
    marginX: 4
  })()}>Test</Button>"
`;

exports[`Adds a comment when a variable is used 1`] = `
"import { Button } from "@react-spectrum/rainbow";

<Button // TODO: update this style prop
marginX={margin}>Test</Button>"
`;

exports[`Bails on UNSAFE_style if it contains a spread 1`] = `
"<div
  // TODO: check this UNSAFE_style
  UNSAFE_style={{
    color: 'var(--spectrum-gray-900)',
    ...otherStyles
  }}>
  Test
</div>"
`;

exports[`Converts UNSAFE_className to className on View 1`] = `
"<div className="hi">
  Test
</div>"
`;

exports[`Handles Flex 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={style({
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    gap: 4,
    flexWrap: "wrap"
  })()}>Test</div>"
`;

exports[`Handles Grid 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={style({
    display: "grid",
    gridTemplateAreas: ["header  header", "sidebar content", "footer  footer"],
    gridTemplateColumns: ["1fr", "3fr"],
    gridTemplateRows: [20, "auto", 20],
    height: "[480px]",
    gap: 2
  })()}>
  Test
</div>"
`;

exports[`Handles UNSAFE_style 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={style({
    color: {
      default: "gray-900",
      isErrorState: "red-900"
    },

    borderWidth: {
      default: 2,

      border: {
        thin: 1
      }
    },

    borderRadius: {
      default: "none",
      isRounded: "default"
    },

    display: "flex",
    gap: 4,
    fontSize: "base",
    fontWeight: "bold",
    gridTemplateAreas: ["foo bar", "baz qux"],
    transitionDuration: 160
  })({
    isErrorState,
    border,
    isRounded: isRounded(props)
  })}>
  Test
</div>"
`;

exports[`Handles View 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={style({
    borderWidth: 1,
    borderColor: "gray-400",
    borderRadius: "lg",
    padding: 5,

    backgroundColor: {
      default: "blue-1000",
      dark: "blue-900"
    },

    flex: 1
  })()}>
  Test
</div>"
`;

exports[`Handles View colorVersion={5} 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={style({
    borderWidth: 1,
    borderColor: "gray-200",
    borderRadius: "lg",
    padding: 5,

    backgroundColor: {
      default: "blue-1000",
      dark: "blue-900"
    }
  })()}>
  Test
</div>"
`;

exports[`Handles View colorVersion={6} 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={style({
    borderWidth: 1,
    borderColor: "gray-200",
    borderRadius: "lg",
    padding: 5,
    backgroundColor: "blue-600"
  })()}>
  Test
</div>"
`;

exports[`Handles dimensions 1`] = `
"import { Button } from "@react-spectrum/rainbow";
import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<Button
  css={style({
    marginX: 4,
    marginY: 5,
    width: "full"
  })()}>Test</Button>"
`;

exports[`Handles renamed components 1`] = `
"import { Button as RSPButton } from "@react-spectrum/rainbow";
import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<RSPButton
  css={style({
    marginX: 4,
    marginY: 5,
    width: "full"
  })()}>Test</RSPButton>"
`;

exports[`Handles responsive style props 1`] = `
"import { TextField } from "@react-spectrum/rainbow";
import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<TextField label="Name" css={style({
  width: {
    default: 40,
    lg: "[400px]"
  }
})()} />"
`;

exports[`Handles responsive style props 2`] = `
"import { TextField } from "@react-spectrum/rainbow";
import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<TextField label="Name" css={style({
  width: {
    default: 40,
    lg: "[400px]"
  }
})()} />"
`;

exports[`Merges UNSAFE_className with macro className 1`] = `
"import { style } from "@react-spectrum/rainbow/style-macro/spectrum-theme" with { type: "macro" };

<div
  className={"hi" + style({
    marginX: 4
  })()}>
  Test
</div>"
`;
